File: Python/_540_singleNonDuplicate.py
from typing import List


class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        n = len(nums)
        l, r = -1, n // 2
        while l + 1 < r:
            mid = (l + r) // 2
            if nums[mid * 2] == nums[mid * 2 + 1]:
                l = mid
            else:
                r = mid
        return nums[r * 2]

File: Java/src/_540_singleNonDuplicate/_540_singleNonDuplicate.java
package _540_singleNonDuplicate;

public class _540_singleNonDuplicate {
}

class Solution {
    public int singleNonDuplicate(int[] nums) {
        int n = nums.length;
        int l = -1, r = n / 2;
        while (l + 1 < r) {
            int mid = (l + r) / 2;
            if (nums[mid * 2] == nums[mid * 2 + 1]) {
                l = mid;
            } else {
                r = mid;
            }
        }
        return nums[r * 2];
    }
}